hacspec-scrambledb/scrambledb/src/join.rs

```diff
@@ -123,49 +99,65 @@ pub fn join_conversion(
     converter_context: &ConverterContext,
     bpk_receiver: BlindingPublicKey,
     ek_receiver: &HpkePublicKey,
-    tables: Vec<BlindTable>,
+    table: Table<BlindedPseudonymizedData>,
     randomness: &mut Randomness,
-) -> Result<Vec<ConvertedTable>, Error> {
-    let mut converted_tables = Vec::new();
+) -> Result<Table<BlindedPseudonymizedData>, Error> {
     let conversion_target = randomness.bytes(SECPAR_BYTES)?.to_owned();
-
-    for table in tables {
-        for blind_column in table.columns() {
-            let attribute = blind_column.attribute();
-
-            let mut converted_data = Vec::new();
-            for (blind_identifier, encrypted_value) in blind_column.data() {
-                let blinded_pseudonymized_datum = BlindedPseudonymizedData {
-                    blinded_handle: BlindedPseudonymizedHandle(blind_identifier),
-                    encrypted_data_value: EncryptedDataValue {
-                        attribute_name: attribute.clone(),
-                        value: encrypted_value,
-                        encryption_level: 1u8,
-                    },
-                };
-                let blinded_pseudonymized_datum = convert_blinded_datum(
-                    &converter_context.coprf_context,
-                    &bpk_receiver,
-                    &ek_receiver,
-                    &conversion_target,
-                    &blinded_pseudonymized_datum,
-                    randomness,
-                )?;
-
-                converted_data.push((
-                    blinded_pseudonymized_datum.blinded_handle.0,
-                    blinded_pseudonymized_datum.encrypted_data_value.value,
-                ));
-            }
-            let mut converted_table_column = Column::new(attribute.clone(), converted_data);
-            converted_table_column.sort();
-            converted_tables.push(ConvertedTable::new(
-                join_identifier(table.identifier()),
-                converted_table_column,
-            ));
-        }
-    }
-    Ok(converted_tables)
+    let mut converted_data = table
+        .data()
+        .iter()
+        .map(|entry| {
+            convert_blinded_datum(
+                &converter_context.coprf_context,
+                &bpk_receiver,
+                ek_receiver,
+                &conversion_target,
+                entry,
+                randomness,
+            )
+        })
+        .collect::<Result<Vec<BlindedPseudonymizedData>, Error>>()?;
+
+    converted_data.sort();
+
+    Ok(Table::new(table.identifier().into(), converted_data))
+    // for table in tables {
```

Comment:
What's with this commented code?

Reply:
Sorry, this is the old code. I'll make a another pass updating the docs on here as well.